code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -75,9 +75,13 @@
 	if( istype(target, /obj/structure/reagent_dispensers/watertank) && flag)
 		var/obj/o = target
 		var/amount = o.reagents.trans_to_obj(src, 50)
-		user << "<span class='notice'>You fill [src] with [amount] units of the contents of [target].</span>"
-		playsound(src.loc, 'sound/effects/refill.ogg', 50, 1, -6)
-		return
+		if((amount == 0))

```

Comment:
You don't need to `((condition))` If checks work just fine like this: `if (amount == 0)`